qcodes/utils/dataset/doNd.py

```diff
@@ -283,11 +282,14 @@ def do1d(
 
         for set_point in tqdm(setpoints, disable=not show_progress):
             param_set.set(set_point)
+            t0 = time.perf_counter()
             datasaver.add_result(
                 (param_set, set_point), *call_param_meas(), *additional_setpoints_data
             )
-
-    param_set.post_delay = original_delay
+            datasaver_time = abs(time.perf_counter() - t0)
```

Comment:
I think max would be more correct than abs 
```suggestion
            datasaver_time = max((time.perf_counter() - t0), 0)
```

In that way if datasaving takes longer than the sleep time we will not sleep at all. With the current code we would sleep the delta between the two again

Reply:
In the following if statement (please pretend you are not seeing the print below :) ), I only sleep the delta value if the provided delay is greater than datasaver_time, otherwise there is no sleep if datasaver_time is bigger than the delay.

Reply:
It's to a certain extend a matter of preference but to me 

```python
datasaver_time = max((time.perf_counter() - t0), 0)
sleep(datasaver_time)
```
is cleaner than the if statement. I don't think there is any signicifant performance difference

Reply:
Agree. This is cleaner.  I will modify accordingly. Thanks